File: Common/Logger/src/common/util/Logger.java
package common.util;

import java.text.SimpleDateFormat;
import java.util.Calendar;
import java.util.GregorianCalendar;

/**
 * Logs information on the console.
 *
 * @author Alexander Komischke
 */
public final class Logger {

    private static final boolean shouldLog = true;
    private static final String separator       = "\t : \t";

    private Logger() {}

    private static String format(String message) {
        if (!shouldLog) {
            return "";
        }
        Calendar         calendar         = new GregorianCalendar();
        SimpleDateFormat simpleDateFormat = new SimpleDateFormat("yyy.MM.dd HH:mm:ss");
        return simpleDateFormat.format( calendar.getTime() ) + separator + message;
    }

    public static String log( Object o ) {
        return log( o.toString() );
    }

    public static String log( Object o, Object argument ) {
        return log( o.toString() + separator + argument.toString() );
    }

    public static String log( final String message) {
        final String m = format(message);
        if (shouldLog) {
            System.out.println(m);
        }
        return m;
    }

    public static String log( String[] messages ) {
        return logStringArray("", messages, "" );
    }

    public static String logStringArray(String prefix, String[] messages, String suffix ) {
        if ( messages == null ) {
            return "";
        }
        StringBuilder s = new StringBuilder();
        for ( String message : messages ) {
            s.append( log( prefix + message + suffix) );
        }
        return s.toString();
    }
}

File: Common/Logger/src/common/util/Logger.kt
package common.util

import java.text.SimpleDateFormat
import java.util.*

/**
 * Logs information on the console.
 *
 * @author Alexander Komischke
 */
object Logger {
    private const val shouldLog = true
    private const val separator = "\t : \t"
    private const val empty = ""
    private const val dateFormat = "yyy.MM.dd HH:mm:ss"

    private infix fun format(message: String): String =
            if (!shouldLog) empty
            else SimpleDateFormat(dateFormat)
                    .format(GregorianCalendar().time) + separator + message

    infix fun log(o: Any?) = log(o.toString())

    fun log(vararg o: Any?) {
        var strings = empty
        for (i in 0 until (o.size)) {
            strings += i.toString() + separator
        }
        log(strings)
    }

    infix fun log(message: String) {
        if (shouldLog) {
            println(this format message)
        }
    }

}
